Modules/Product/Http/Services/ProductService.php

```diff
@@ -103,12 +110,114 @@ public function fetch($request): array
+            $productVariations = Product::find($v->id)->productVariation()->where('status', '1')->where('stock','>', '0')->get();
+            $productVariationsCount = $productVariations->count();
+            if ($productVariationsCount > 0) {
+                $variantMinPrice = Product::find($v->id)->productVariation()->min('price');
+                $price = $variantMinPrice . '+';
+                $variantStock = Product::find($v->id)->productVariation()->sum('stock');
+                $availability = $variantStock > 0 ? 'in stock' : 'out of stock';
+                $productStock += $variantStock;
+            } else {
+                $price = $v->usd_wholesale_price;
+                $availability = $v->stock > 0 ? 'in stock' : 'out of stock';
+            }
+            $variations=[];
+            if(!empty($productVariations)) {
+                foreach ($productVariations as $pv) {
+                    $image = !empty($pv->image) ? $pv->image : $v->featured_image;
+                    $sku = !empty($pv->sku) ? $pv->sku : $v->sku;
+                    $stock = !empty($pv->stock) ? $pv->stock : $v->stock;
+                    $variableArr = [];
+                    if (!empty($pv->value1)) {
+                        $variableArr[] = $pv->value1;
+                    }
+                    if (!empty($pv->value2)) {
+                        $variableArr[] = $pv->value2;
+                    }
+                    if (!empty($pv->value3)) {
+                        $variableArr[] = $pv->value3;
+                    }
+                    $variations[] = array(
+                        'id' => $v->id,
+                        'usd_wholesale_price' => $pv->price,
+                        'usd_retail_price' => $pv->retail_price,
+                        'variant_id' => $pv->id,
+                        'variant' => implode('/', $variableArr),
+                        'name' => $pv->name,
+                        'sku' => $sku,
+                        'featured_image' => $image,
+                        'stock' => $stock
+                    );
+                }
+            }
+            $prepacks  = [];
+            $productPrepackages = Product::find($v->id)->productPrepack()->where('active', '1')->get();
+            if (!empty($productPrepackages)) {
+                foreach ($productPrepackages as $key => $var) {
+                    $prepacks[] = array(
+                        'id' => $var->id,
+                        'style' => $var->style,
+                        'pack_name' => $var->pack_name,
+                        'size_ratio' => $var->size_ratio,
+                        'size_range' => $var->size_range,
+                        'packs_price' => $var->packs_price,
+                        'active' => $var->active,
+                        'created_at' => $var->created_at,
+                        'updated_at' => $var->updated_at,
+                        'variationWishId' => ''
+                    );
+
+                }
```

Comment:
You can avoid this loop completely in Product query like this:
$products = Product::with(['productPrepack' => function ($query) {
    $query->select('id', 'name');
}])->get();
This way the relational data will only contain id and name, you can get all the necessary colums and once you have you can simply convert those to array like
foreach($product as $product){
$prepacks = $prodcut->productPrepack()->toArray();
}
NO need for this loop,
NOTE: you can confirm the syntax etc for the code i'm giving you.